redskyctl/internal/commands/run/command.go

```diff
@@ -0,0 +1,377 @@
+	// Wait for the controller to become ready if it is installed
+	checkOpts := check.ControllerOptions{
+		Config:    o.Config,
+		IOStreams: discard,
+	}
+	if err := checkOpts.CheckController(ctx); err == nil {
+		return o.checkControllerVersion()
+	}
+
+	// Error indicates the controller is not yet installed
+	initOpts := &initialize.Options{
+		GeneratorOptions: initialize.GeneratorOptions{
+			Config:               o.Config,
+			IOStreams:            discard,
+			IncludeBootstrapRole: true,
+			Image:                o.initializationModel.ControllerImage,
+		},
+		Wait: true,
+	}
+	if err := initOpts.Initialize(ctx); err != nil {
+		return err
+	}
+
+	// Now that we are installed, wait for it to become ready again
+	checkOpts.Wait = true
+	if err := checkOpts.CheckController(ctx); err != nil {
+		return err
+	}
+
+	return o.checkControllerVersion()
+}
+
+func (o *Options) listKubernetesNamespaces() tea.Msg {
+	ctx := context.TODO()
+	msg := internal.KubernetesNamespacesMsg{}
+
+	cmd, err := o.Config.Kubectl(ctx, "get", "namespaces", "--output", "name")
+	if err != nil {
+		return err
+	}
+	data, err := cmd.Output()
+	if err != nil {
+		return err
+	}
+	scanner := bufio.NewScanner(bytes.NewReader(data))
+	for scanner.Scan() {
+		if ns := strings.TrimPrefix(scanner.Text(), "namespace/"); !o.hideKubernetesNamespace(ns) {
+			msg = append(msg, ns)
+		}
+	}
+
+	return msg
+}
+
+func (o *Options) listStormForgerTestCaseNames() tea.Msg {
+	ctx := context.TODO()
+	msg := internal.StormForgerTestCasesMsg{}
+
+	orgs, err := forge(ctx, "organization", "list")
+	if err != nil {
+		return nil
+	}
+
+	for i := range orgs.Data {
+		org := orgs.Data[i].Attributes.Name
+		testCases, err := forge(ctx, "test-case", "list", org)
+		if err != nil {
+			return err
+		}
+		for j := range testCases.Data {
+			testCase := testCases.Data[j].Attributes.Name
+			msg = append(msg, fmt.Sprintf("%s/%s", org, testCase))
+		}
+	}
+
+	return msg
+}
+
+func (o *Options) generateExperiment() tea.Msg {
+	msg := internal.ExperimentMsg{}
+
+	o.Generator.Application.Default()
+	o.Generator.SetDefaultSelectors()
+
+	if err := o.Generator.Execute(&msg); err != nil {
+		return err
+	}
+
+	return msg
+}
+
+func (o *Options) createExperimentInCluster() tea.Msg {
+	ctx := context.TODO()
+
+	data, err := kio.StringAll(o.runModel.experiment)
+	if err != nil {
+		return err
+	}
+
+	cmd, err := o.Config.Kubectl(ctx, "create", "-f", "-")
+	if err != nil {
+		return err
+	}
+
+	cmd.Stdin = strings.NewReader(data)
+	if _, err := cmd.Output(); err != nil {
+		return fmt.Errorf("could not create experiment, %w", err)
+	}
+
+	return internal.ExperimentCreatedMsg{}
+}
+
+func (o *Options) createExperimentOnScreen() tea.Msg {
+	content, err := kio.StringAll(o.runModel.experiment)
+	if err != nil {
+		return err
+	}
+	o.previewModel.Preview.SetContent(content)
+	return nil
+}
+
+func (o *Options) refreshTrials() tea.Msg {
+	ctx := context.TODO()
+
+	// TODO Where do we get the namespace/selector from?
```

Comment:
Is this todo still a concern?

Reply:
Yes, I'll update the comment so it makes sense to humans who aren't me.